nemo/core/optim/mcore_optim.py

```diff
@@ -55,8 +55,13 @@ def state_dict(self):
     def load_state_dict(self, state_dict):
         self.mcore_optimizer.load_state_dict(state_dict)
 
-    def sharded_state_dict(self, model_sharded_state_dict, is_loading: bool = False, **kwargs):
-        return self.mcore_optimizer.sharded_state_dict(model_sharded_state_dict, is_loading, **kwargs)
+    def sharded_state_dict(
+        self, model_sharded_state_dict, optimizer_state_dict=None, is_loading=False, dist_ckpt_parallel_save=False
+    ):
+        sharding_type = 'fully_sharded_bucket_space' if dist_ckpt_parallel_save else 'dp_zero_gather_scatter'
```

Comment:
@akoumpa I'm working on an improvement of the DistOpt ckpt format.
The new `fully_sharded_model_space` format will be ready in 1-2 weeks and will deprecate the `fully_sharded_bucket_space` one.
When are you planning to merge this MR? Do you think we can wait for the new format to land, so that we don't have problems with deprecating the old format?

Reply:
How about we comment line 61 and hard-code `dp_zero_gather_scatter` and in two weeks uncomment it with the latest distopt format?

Reply:
I'm good with that :+1: